src/vue/pages/Trade.vue

```diff
@@ -0,0 +1,265 @@
+      const quoteAsset = value.split('/')[1]
+      this.assets.base = baseAsset
+      this.assets.quote = quoteAsset
+      // pass assets to child router-view components
+      this.childRoutesConfig.assets.base = baseAsset
+      this.childRoutesConfig.assets.quote = quoteAsset
+
+      const accountBalances = this.accountBalances
+      const baseBalance = accountBalances.find(i => i.asset === baseAsset)
+      const quoteBalance = accountBalances.find(i => i.asset === quoteAsset)
+      this.balances.baseBalance = baseBalance.balance
+      this.balances.quoteBalance = quoteBalance.balance
+
+      this.$router.replace({
+        name: this.$route.name,
+        query: {
+          base: baseAsset,
+          quote: quoteAsset,
+        },
+      })
+    },
+  },
+  async created () {
+    await this.loadBalances()
+    await this.loadTradablePairs()
+  },
+  methods: {
+    ...mapActions({
+      loadBalances: vuexTypes.LOAD_ACCOUNT_BALANCES_DETAILS,
+    }),
+    async loadTradablePairs () {
+      try {
+        const response = await Sdk.horizon.assetPairs.getAll()
+        this.formatTradablePairs(response.data)
+        this.setDefaultSelectedPair(response.data)
+      } catch (error) {
+        ErrorHandler.process(error)
+      }
+    },
+    formatTradablePairs (pairs) {
+      this.formattedPairs = pairs.map(item => {
+        return `${item.base}/${item.quote}`
+      })
+    },
+    setDefaultSelectedPair (pairs) {
+      const queryBase = this.$route.query.base
+      const queryQuote = this.$route.query.quote
+
+      // if the user comes on Trade page from another place with some base and
+      // quote query params inside URL, we check that they're correct
+      // (exists in system) and set the appropriate
+      if (this.isQueryParamsValid(pairs)) {
+        const pair = pairs.find((i) => {
+          return i.base === queryBase &&
+                  i.quote === queryQuote
+        })
+        this.selectedPair = `${pair.base}/${pair.quote}`
+      } else {
+        this.selectedPair = this.formattedPairs[0]
+        if (queryBase && queryQuote) {
+          Bus.error(globalize('trade.error-invalid-base-quote-query-in-link', {
+            invalidBase: queryBase,
+            invalidQuote: queryQuote,
+            defaultPair: this.formattedPairs[0],
+          }))
+        }
+      }
+    },
+    isQueryParamsValid (pairs) {
+      const queryBase = this.$route.query.base
+      const queryQuote = this.$route.query.quote
+
+      if (queryBase && queryQuote) {
+        return Boolean(pairs.find((i) => {
+          return i.base === queryBase && i.quote === queryQuote
+        }))
+      }
+      return false
+    },
+    closeBuyOrderDrawer () {
+      this.isCreateBuyOrderDrawerShown = false
+      this.childRoutesConfig.isNeededToReloadData = true
+    },
+    closeSellOrderDrawer () {
+      this.isCreateSellOrderDrawerShown = false
+      this.childRoutesConfig.isNeededToReloadData = true
```

Comment:
let’s use events for things like reload data requests

Reply:
I pass this value as the prop to child components. How can I use events to pass data from parent to child? For parent-child architecture, we could use two ways (really dirty):
`this.$parent.$on('event', handler)` in children component to listen parent component **OR**
`this.$refs.childComponent.handler()` from parent.
But we have the parent component with `router-view`, so we can not use these ways. So we can use props or route params. In my opinion, the props is better. If I lost some other option, tell me, please.

Reply:
just call the method that fetches the new data instead of toggling of isNeededToReloadData